test/log.ts

```diff
@@ -413,439 +398,275 @@ describe('Log', () => {
+      const entries = entriesFactory({
+        message: 'hello world'.padEnd(2000, '.'),
+        metadata: {
+          stack: 'hello world'.padEnd(2000, '.'),
+        },
       });
 
-      it('should pass correct arguments to write', async () => {
-        const assignedEntries = [] as Entry[];
-
-        assignSeverityToEntriesOverride = () => {
-          return assignedEntries;
-        };
+      log.maxEntrySize = maxSize;
+      log.truncateEntries(entries);
 
-        log.write = async (entry: Entry, labels: WriteOptions) => {
-          assert.strictEqual(entry, assignedEntries);
-          assert.strictEqual(labels, LABELS);
-        };
-
-        await log.info(ENTRY, LABELS);
-      });
+      const message: string = entries[0].jsonPayload!.fields!.message
+        .stringValue!;
+      const stack: string = entries[0].jsonPayload!.fields!.metadata
+        .structValue!.fields!.stack.stringValue!;
+      assert.strictEqual(stack, '');
+      assert.ok(message.startsWith('hello world'));
+      assert.ok(message.length < maxSize + entryMetaMaxLength);
     });
+  });
 
-    describe('notice', () => {
-      it('should format the entries', async () => {
-        assignSeverityToEntriesOverride = (
-          entries: Entry[],
-          severity: string
-        ) => {
-          assert.strictEqual(entries, ENTRY);
-          assert.strictEqual(severity, 'NOTICE');
-        };
-
-        await log.notice(ENTRY, LABELS);
-      });
-
-      it('should pass correct arguments to write', async () => {
-        const assignedEntries = [] as Entry[];
-
-        assignSeverityToEntriesOverride = () => {
-          return assignedEntries;
-        };
-
-        log.write = async (entry: Entry, labels: WriteOptions) => {
-          assert.strictEqual(entry, assignedEntries);
-          assert.strictEqual(labels, LABELS);
-        };
-
-        await log.notice(ENTRY, LABELS);
-      });
-    });
+  describe('severity shortcuts', () => {
+    let ENTRY: Entry;
+    let LABELS: WriteOptions;
+    let assignSeverityStub: sinon.SinonStub;
+    let writeStub: sinon.SinonStub;
 
-    describe('warning', () => {
-      it('should format the entries', async () => {
-        assignSeverityToEntriesOverride = (
-          entries: Entry[],
-          severity: string
-        ) => {
-          assert.strictEqual(entries, ENTRY);
-          assert.strictEqual(severity, 'WARNING');
-        };
+    beforeEach(() => {
+      ENTRY = {} as Entry;
+      LABELS = [] as WriteOptions;
+      assignSeverityStub = sinon.stub(Log, 'assignSeverityToEntries_');
+      writeStub = sinon.stub(log, 'write');
+    });
+
+    afterEach(() => {
+      assignSeverityStub.restore();
+      writeStub.restore();
+    });
+
+    [
+      'alert',
+      'critical',
+      'debug',
+      'emergency',
+      'error',
+      'info',
+      'notice',
+      'warning',
+    ].forEach(severityMethodName => {
+      describe(severityMethodName, () => {
```

Comment:
@bcoe are you OK with these dynamic `describe`s? the original tests for the various severity shortcuts all tested the same thing with the only difference being the severity name.